### PR TITLE
issues/1 add repo template v1

### LINKED ISSUE #1: repository template v1 for all RoboSter repositories
### Problem

In order not to repeat same actions every time we create repositories
As a RoboSter team member
I want to have repository template that will utilize all the open source standard requirements for any new repository

### Solution

Repository should provide

* simple `README.md` template
* `.gitignore`
* Contribution Guidelines
* Apache 2.0 License
* Issues & Pull Request templates

### PR DESCRIPTION
**Pull request by katunilya**
### Log

* add simple template `README.md`
* add Apache 2.0 License 
* add Contribution Guidelines (awaiting discussion)
* add issue templates
	* feature request
	* groundwork
	* bug report
* add standard code of conduct
* add (empty) .gitignore 

### To Do

- [x] GitHub actions WIP check 
- [x] .gitignore for common folders & files
- [x] add more details on setting up pull request in `CONTRIBUTING.md`
- [x] add separated paragraph for hardware info in bug report issue template

### CODE REVIEW

**Comment by bymse** [comment]
check out issue, I've added a comment there
and it'll be a good practice to link issue to the pr
image

**Comment by katunilya** [comment]
Worth mentioning:

1. When I used repository templates on GitHub for the last time they had no support for transferring labels
2. If one wants to clone repository than you'd better used some client (I recommend [Fork](https://git-fork.com/) (which is powerful) or [GitHub Desktop](https://desktop.github.com/) (which is simple) as you'll bump into ssh keys issues

**Comment by katunilya** [comment]
@bymse intended to write `closed #1` when merging as it works so as well

File: README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Template
+
+## Description
+
+Please provide consistent description of your repository. Explain what you develop and why it's better than alternatives (if there are any)
```
Comment (bymse) [inline README.md:5]: It's better to just provide link to common wiki
Without any additional explanations
ause it's better to store info in one place

**Comment by katunilya** [comment]
@bymse 

>issues
>I suggest those types:
> * feature, as general issue for large improvement of functionality
> * task, simple issue for 1-3 commits, for example: new api method for some currently available info
> * infra, for infrastructure improvements

**Groundwork** utilizes task & infra and currently I don't see any reason for strong separation

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# List files, folders & patterns that git should ignore
+# Remember that no binaries or user-specific files should be tracked
+
```
Comment (bymse) [inline .gitignore:3]: let's put here:
.idea
.vs*
[Dd]ebug/
[Dd]ebugPublic/
[Rr]elease/
[Rr]eleases/
x64/
x86/
build/
bld/
[Bb]in/
[Oo]bj/
[Oo]ut/

**Comment by bymse** [comment]
> **Groundwork** utilizes task & infra

Task has greater priority than any infra, but it's  not so important for now

File: README.md (hunk 1 above)
Comment (katunilya) [reply README.md:5]: I find it important providing _short_ description in readme & of course leave link to Wiki

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## Log
+
+Provide a list of changes and describe used solutions referencing issues
```
Comment (bymse) [inline .github/pull_request_template.md]: well, imho, you can check changes in commits section

Comment (katunilya) [reply .github/pull_request_template.md]: only if commit history is good and it's usually not

File: .gitignore (hunk 2 above)
Comment (katunilya) [reply .gitignore:3]: Intentionally left empty to sum up requests in one file

I would also add

```
# Visual Studio Code
.vscode
# F# package for VS Code
.ionide
# Prefere to compile C++ code by hands with MinGW and usually get executable in the same folder
*.exe
```

File: .github/pull_request_template.md (hunk 3 above)
Comment (bymse) [reply .github/pull_request_template.md]: and it will be the first review-comment, that commits history should be recreated

Comment (katunilya) [reply .github/pull_request_template.md]: Still there is nothing bad in keeping all changes on top of the page as they can be mixed among comments? And it helps other users to grasp information quicker imho

Comment (bymse) [reply .github/pull_request_template.md]: But it's another thing that you have to do/to remember
And it's not so important if you have good commits history

Comment (katunilya) [reply .github/pull_request_template.md]: It’s still easier than refining commit history (which still should be done) & I believe we can decide if we need this by voting in our chat

**Comment by katunilya** [comment]
Add WIP check via GitHub App [WIP](https://github.com/marketplace/wip)

**Comment by katunilya** [comment]
**Solution for label issue**

We can define in project settings all the labels that will be added to the new repository whenever we create one

**Comment by bymse** [comment]
> Add WIP check via GitHub App [WIP](https://github.com/marketplace/wip)

Let's also add this feature https://help.github.com/en/github/administering-a-repository/enabling-required-status-checks

**Comment by katunilya** [comment]
Good idea, but it will duplicate checks (as I understand). If that's alright than OK

**Comment by bymse** [comment]
Not sure about `duplicate checks`
but even so, I don't think that it's a problem

**Comment by katunilya** [comment]
Ok.

Currently we have WIP installed as app, so there is no way to include it into branch merging rules. I'll add separated `wip.yml` to setup it as GitHub Action.

Still I'm not sure if repositories inherit branch merging rules, but we will try

File: .gitignore (hunk 2 above)
Comment (OlegMoiseev) [reply .gitignore:3]: I would add:
```
# Virtual environment for Python
venv/
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+3. Scroll down to '....'
+4. See error
+
+### Expected Behavior
+
+A clear and concise description of what you expected to happen.
+
+### Additional Context
+
+Add any other context about the problem here. For example insert there device data - model, OS, Browser, versions of software, screenshots, etc.
```
Comment (OlegMoiseev) [inline .github/ISSUE_TEMPLATE/bug_report.md:32]: IMHO, we should add "device data" into important paragraphs to easier reproducing bugs.

**Review by OlegMoiseev — CHANGES_REQUESTED** [review]
In general OK.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 4 above)
Comment (katunilya) [reply .github/ISSUE_TEMPLATE/bug_report.md:32]: Got it. I took this from some old template for projects without hardware oriented problems. Will fix.

**Review by AlexeyRyb — APPROVED** [review]

**Review by KiriyMIke — APPROVED** [review]

**Review by ilyatyutko — APPROVED** [review]

**Review by LazyDareDevil — APPROVED** [review]

**Review by OlegMoiseev — APPROVED** [review]
All ok!

**Review by bymse — APPROVED** [review]
Cool. 
Then we should check how all template-repo settings will be applied to new repos

**Pull request merged**